include/mender-utils.h

```diff
@@ -91,6 +91,7 @@ typedef enum {
     MENDER_FAIL            = -1, /**< Failure */
     MENDER_NOT_FOUND       = -2, /**< Not found */
     MENDER_NOT_IMPLEMENTED = -3, /**< Not implemented */
+    MENDER_STATE_LOOP      = -4, /**< State loop */
```

Comment:
It is an error, I would name it with something more explicit. Examples:
* `MENDER_STATE_LOOP_DETECTED` (this is similar to the C++ error name)
* `MENDER_MAX_STATE_COUNT_EXCEEDED`
* ...

Reply:
Also, I don't really like exposing this error in `mender-utils.h`. But this is kind of the issue with no proper namespaces as the project is structured :cry: